src/validations/register.validation.js

```diff
@@ -0,0 +1,33 @@
+import Joi from "joi";
+import { joiPasswordExtendCore } from "joi-password";
+
+const joiPassword = Joi.extend(joiPasswordExtendCore);
+
+const validateRegister = (req, res, next) => {
+  const registerSchema = Joi.object().keys({
+    firstname: Joi.string().min(3).trim().required(),
+    lastname: Joi.string().min(3).trim().required(),
+    email: Joi.string().email().lowercase().trim().required(),
+    phone: Joi.string().min(10).trim(),
+    password: joiPassword
+      .string()
+      .min(8)
+      .max(16)
+      .minOfSpecialCharacters(1)
+      .minOfLowercase(1)
+      .minOfUppercase(1)
+      .minOfNumeric(1)
+      .noWhiteSpaces()
+      .trim()
+      .required(),
+  });
```

Comment:
Why do you need all those checks for passwords? Try to minimize them.

Reply:
So, the registration feature requires an Alphanumeric Password and that's the only way I know to validate the input.